stablehlo/dialect/TypeInference.cpp

```diff
@@ -2440,18 +2552,34 @@ LogicalResult inferGatherOp(
                           /*operandShape=*/operandShape,
                           /*startIndicesShape=*/startIndicesShape,
                           /*sliceSizesShape=*/sliceSizesShape, offsetDims,
-                          collapsedSliceDims, startIndexMap, indexVectorDim)))
+                          collapsedSliceDims, operandBatchingDims,
+                          startIndicesBatchingDims, startIndexMap,
+                          indexVectorDim)))
     return failure();
 
-  // gather_c8
-  for (auto dim : collapsedSliceDims) {
-    int64_t sliceDimSize = sliceSizes[dim];
-    if (sliceDimSize > 1)
-      return emitOptionalError(location, "slice_sizes collapsed dimension ",
-                               dim, " should <= 1 but got ", sliceDimSize);
+  auto checkSliceSizesOne = [&](ArrayRef<int64_t> dims, StringRef name) {
```

Comment:
The message here is a bit confusing, do you have a better way to say that the slice size should be 1 because it's a collapsed/batching dim?

Reply:
Updated. wdyt?

Reply:
How about:
```
return emitOptionalError(location, "Expects that for each dim in ", name,
                         ", slice_sizes[dim] should be <=1, but got ",
                         sliceDimSize);
```

Reply:
SG!